src/action-sheets/index.js

```diff
@@ -171,6 +172,13 @@ const muteTopic = async ({ auth, streamId, topic, streams }) => {
 muteTopic.title = 'Mute topic';
 muteTopic.errorMessage = 'Failed to mute topic';
 
+const copyLinkToTopic = async ({ auth, streamId, topic, streams }) => {
+  const topicURL = getStreamTopicURL(auth.realm, streamId, topic, streams);
+  Clipboard.setString(topicURL.toString());
+};
+copyLinkToTopic.title = 'Copy link to topic';
+copyLinkToTopic.errorMessage = 'Failed to copy topic link';
```

Comment:
These strings will get translated, so they should be added to `messages_en.json` so that we can upload them to Transifex and get translations for them. (I realize that the fact these are translated isn't obvious here -- it happens at a bit of a distance, lower down in this file. Search for the string `_(` and look in `showTopicActionSheet`.)

For general context on this workflow, see [howto/translations.md](https://github.com/zulip/zulip-mobile/blob/777b584a6b066a5daeedb872f252fab33708f73f/docs/howto/translations.md#all-code-contributors-wiring-our-code-for-translations).